devolo_home_control_api/backend/mprm_rest.py

```diff
@@ -71,11 +71,95 @@ def get_name_and_element_uids(self, uid: str):
         """
         data = {"method": "FIM/getFunctionalItems",
                 "params": [[uid], 0]}
-        response = self.post(data)
+        response = self._post(data)
         self._logger.debug(f"Response of 'get_name_and_element_uids':\n{response}")
         return response["result"]["items"][0]["properties"]
 
-    def post(self, data: dict) -> dict:
+    def refresh_session(self):
+        """
+        Refresh currently running session. Without this call from time to time especially websockets will terminate.
+        """
+        self._logger.debug("Refreshing session.")
+        data = {"method": "FIM/invokeOperation",
+                "params": [f"devolo.UserPrefs.{self._mydevolo.uuid()}", "resetSessionTimeout", []]}
+        self._post(data)
+
+    def set_binary_switch(self, uid: str, state: bool) -> bool:
+        """
+        Set a binary switch state of a device.
+
+        :param uid: Element UID, something like devolo.BinarySwitch:hdm:ZWave:CBC56091/24
+        :param state: True if switching on, False if switching off
+        :return: True if successfully switched, false otherwise
+        """
+        data = {"method": "FIM/invokeOperation",
+                "params": [uid, "turnOn" if state else "turnOff", []]}
+        response = self._post(data)
+        if response['result']['status'] == 1:
```

Comment:
Checking the response status can be wrapped into a function as it looks the same everytime (almost)

Reply:
Done